src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/ViewCodeCommandHandler.cs

```diff
@@ -53,9 +61,9 @@ public CommandState GetCommandState(ViewCodeCommandArgs args)
 
         public bool ExecuteCommand(ViewCodeCommandArgs args, CommandExecutionContext executionContext)
         {
-            if (TryGetCSharpFilePath(args.SubjectBuffer, out var cSharpFilePath))
+            if (TryGetCSharpFilePath(args.SubjectBuffer, out var csharpFilePath))
             {
-                VsShellUtilities.OpenDocument(ServiceProvider.GlobalProvider, cSharpFilePath);
+                _joinableTaskContext.Factory.Run(() => _documentInteractionManager.OpenDocumentAsync(csharpFilePath, CancellationToken.None));
```

Comment:
Was not running on the UI thread before causing issues?

Reply:
JTF.Run does not mean switch to the UI thread, it just allows you to run async code from sync code and worry less about deadlocks.

This code would all be running on the UI thread by virtue of being a command handler, but if it wasn't, then inside the JTF.Run you would do `await JTF.SwitchToMainThread()` to get there.